magento/pwa-studio
# [PWA-2227] Use sample data with personalized content

## Description

Front-end part for new personalized content module that generates dynamic blocks and cart rule related to a customer segment
Add discount price label in the cart + add styles for dynamic blocks banner on homepage

Test / Demo environment : https://jeremycharron-o4saqja-4txwfa4z2sor6.eu-4.magentosite.cloud/
Back-end pull-requests : https://github.com/magento-commerce/venia-sample-data-modules/pull/19 and https://github.com/magento-commerce/venia-sample-data-modules-ee/pull/2

## Related Issue

Closes PWA-2227

## Acceptance





### Verification Stakeholders

Project maintainers

### Specification

N/A

## Verification Steps






#### Test scenario(s) for direct fix/feature

1. As a guest, verify that "Get Connected" black banner is displayed on the homepage
2. As a guest, add an item to cart and verify that LOYAL10 coupon code cannot be applied in cart.
3. Log in as sharon@example.com sample user and verify that "Welcome back" blue banner is displayed on the homepage
4. As Sharon, verify that "Get Connected" banner is not displayed on homepage
5. As Sharon, verify that LOYAL10 coupon code can be applied in cart and gives 10% off order
6. After testing step 5, verify that discount label is "Frequent Shopper 10% discount"
7. Try to install new sample data on fresh Adobe Commerce environment
8. Verify that new dynamic blocks, customer segments and cart rule are editable in Admin panel.

#### Test scenario(s) for any existing impacted features/areas

1. Verify that setup:upgrade is working and not generating any error

#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?

Yes (Chrome, Safari, Firefox, Edge) + mobile

#### Any ad-hoc/edge case scenarios that need to be considered?

Make sure that wysiwyg/welcome-back background images are being created in /pub directory

## Screenshots / Screen Captures (if appropriate)

On PWA Front-End (you need `jcharron/PWA-2227-use-sample-data-with-personalized-content` on front-end) : 
![black-blue](https://user-images.githubusercontent.com/25036022/148094051-51ad938d-252a-4cbb-bc0f-4d21fa85c757.PNG)
![black-banner](https://user-images.githubusercontent.com/25036022/148094053-a0de62b2-a8bc-4c15-b5f2-bfe09fa7ae4d.PNG)


## Breaking Changes (if any)

No breaking change

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
